fabric/src/main/java/net/minescript/common/Minescript.java

```diff
@@ -3981,6 +4010,84 @@ private static Optional<String> handleScriptFunction(
         }
         return Optional.of(toJsonString(getScreenName().orElse(null)));
 
+      case "container_get_items": // List of Items in Chest
+      {
+        if (!args.isEmpty()){
+          throw new IllegalArgumentException("Expected no params but got: " + argsString);
+        }
+        Screen screen = minecraft.currentScreen;
+        if (screen instanceof HandledScreen<?>) {
+          HandledScreen handledScreen = (HandledScreen) screen;
+          ScreenHandler screenHandler = handledScreen.getScreenHandler();
+          Slot[] items = screenHandler.slots.toArray(new Slot[0]);
+          StringBuilder resultString = new StringBuilder("[");
+          for (int i = 0; i < items.length; i++) {
+            ItemStack itemStack = items[i].getStack();
+            if (itemStack.isEmpty()) {
+              continue;
+            }
+            resultString.append(itemStackToJsonString(itemStack, OptionalInt.of(items[i].id), false));
+            if (i < items.length - 1) {
+              resultString.append(",");
```

Comment:
I tried patching this commit and I get this error with `container_get_items()`:

```
[Render thread/INFO] (ChatHudMixin) [System] [CHAT] JSON error in: {"fcid": 1001, "retval": [{"item": "yellow_concrete", "count": 1, "slot": 0},{"item": "grass_block", "count": 1, "slot": 0},{"item": "blue_concrete", "count": 1, "slot": 0},{"item": "chest", "count": 1, "slot": 0},{"item": "ender_chest", "count": 1, "slot": 0},], "conn": "close"}
```

The issue appears to be the trailing comma between the closing curly brace and closing square bracket: `[{...},]`

That's probably because you're appending a comma after all but the last of the items, but the enclosing loop can skip items via the `continue` above.

In general we shouldn't be constructing the JSON strings ourselves (I'm guilty of that elsewhere). But the quick fix here is to add the comma at the top of the loop if `resultString.length() > 1` (i.e. the `StringBuilder` contains more than just `"["`). See handling of `result` in `entitiesToJsonString(...)` for an example.